File: csharp/Svix/Models/BackgroundTaskFinishedEvent.cs
// this file is @generated
using System.Text;
using Newtonsoft.Json;

namespace Svix.Models
{
    /// <summary>
    /// Sent when a background task is finished.
    /// <summary>
    public class BackgroundTaskFinishedEvent
    {
        [JsonProperty("data", Required = Required.Always)]
        public BackgroundTaskFinishedEvent2 Data { get; set; }

        [JsonProperty("type", Required = Required.Always)]
        public string Type { get; set; }

        public override string ToString()
        {
            StringBuilder sb = new StringBuilder();

            sb.Append("class BackgroundTaskFinishedEvent {\n");
            sb.Append("  Data: ").Append(Data).Append('\n');
            sb.Append("  Type: ").Append(Type).Append('\n');
            sb.Append("}\n");
            return sb.ToString();
        }
    }
}

File: php/src/Models/BackgroundTaskFinishedEvent.php
<?php

// this file is @generated
declare(strict_types=1);

namespace Svix\Models;

/** Sent when a background task is finished. */
class BackgroundTaskFinishedEvent implements \JsonSerializable
{
    private array $setFields = [];

    private function __construct(
        public readonly BackgroundTaskFinishedEvent2 $data,
        public readonly string $type,
        array $setFields = [],
    ) {
        $this->setFields = $setFields;
    }

    /**
     * Create an instance of BackgroundTaskFinishedEvent with required fields.
     */
    public static function create(
        BackgroundTaskFinishedEvent2 $data,
        string $type,
    ): self {
        return new self(
            data: $data,
            type: $type,
            setFields: ['data' => true, 'type' => true]
        );
    }

    public function jsonSerialize(): mixed
    {
        $data = [
            'data' => $this->data,
            'type' => $this->type];

        return \Svix\Utils::newStdClassIfArrayIsEmpty($data);
    }

    /**
     * Create an instance from a mixed obj.
     */
    public static function fromMixed(mixed $data): self
    {
        return new self(
            data: \Svix\Utils::deserializeObject($data, 'data', true, 'BackgroundTaskFinishedEvent', [BackgroundTaskFinishedEvent2::class, 'fromMixed']),
            type: \Svix\Utils::getValFromJson($data, 'type', true, 'BackgroundTaskFinishedEvent')
        );
    }

    /**
     * Create an instance from a json string.
     */
    public static function fromJson(string $json): self
    {
        $data = json_decode(json: $json, associative: true, depth: 512, flags: JSON_THROW_ON_ERROR);

        return self::fromMixed($data);
    }
}
